File: project1.3/project1.3/project1/backend/app/schemas/knowledge.py
from pydantic import BaseModel


class KnowledgeNodeUpdate(BaseModel):
    name: str | None = None
    sort_order: int | None = None
    definition: str | None = None
    key_terms: list[str] | None = None
    teaching_emphasis: str | None = None


class KnowledgeNodeOut(BaseModel):
    id: int
    material_id: int
    parent_id: int | None
    sort_order: int
    name: str
    node_type: str
    definition: str | None
    key_terms: list | None
    teaching_emphasis: str | None
    children: list["KnowledgeNodeOut"] = []

    model_config = {"from_attributes": True}

File: frontend/src/api/knowledge.ts
import type { KnowledgeNode } from '../types'
import { api } from './client'

export async function extractTree(materialId: number): Promise<{ nodes_created: number }> {
  const res = await api.post(`/knowledge/extract/${materialId}`)
  return res.data
}

export async function getTree(materialId: number): Promise<KnowledgeNode[]> {
  const res = await api.get<KnowledgeNode[]>(`/knowledge/tree/${materialId}`)
  return res.data
}

export async function updateNode(
  nodeId: number,
  data: Partial<Pick<KnowledgeNode, 'name' | 'definition' | 'key_terms' | 'teaching_emphasis' | 'solution_steps' | 'sort_order'>>,
): Promise<void> {
  await api.put(`/knowledge/nodes/${nodeId}`, data)
}

export async function deleteTree(materialId: number): Promise<void> {
  await api.delete(`/knowledge/tree/${materialId}`)
}
